src/lua/tweaks.c

```diff
@@ -0,0 +1,135 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "lua/tweaks.h"
+
+#include <lua.h>
+#include <lauxlib.h>
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "core/tweaks.h"
+#include "diag.h"
+#include "lua/utils.h"
+#include "trivia/util.h"
+
+/**
+ * Pushes a tweak value to Lua stack and returns the number of values pushed.
+ */
+static int
+luaT_push_tweak_value(struct lua_State *L, const struct tweak_value *v)
+{
+	switch (v->type) {
+	case TWEAK_VALUE_BOOL:
+		lua_pushboolean(L, v->bval);
+		return 1;
+	case TWEAK_VALUE_INT:
+		lua_pushinteger(L, v->ival);
+		return 1;
+	case TWEAK_VALUE_STR:
+		lua_pushstring(L, v->sval);
+		return 1;
+	default:
+		unreachable();
+	}
+}
+
+/**
+ * Looks up a tweak value by name (2nd arg) and pushes it to Lua stack.
+ * Returns nothing if there's no such tweak.
+ */
+static int
+luaT_tweaks_index(struct lua_State *L)
+{
+	const char *name = luaL_checkstring(L, 2);
+	struct tweak *t = tweak_find(name);
+	if (t == NULL)
+		return 0;
```

Comment:
Nit: I would `lua_pushnil()` + `return 1` to mimic a table behavior more precisely.

Reply:
Makes sense. Fixed.

<details><summary>diff</summary>

```diff
diff --git a/src/lua/tweaks.c b/src/lua/tweaks.c
index 8a3c8045fcc8..40342652fc15 100644
--- a/src/lua/tweaks.c
+++ b/src/lua/tweaks.c
@@ -38,15 +38,17 @@ luaT_push_tweak_value(struct lua_State *L, const struct tweak_value *v)
 
 /**
  * Looks up a tweak value by name (2nd arg) and pushes it to Lua stack.
- * Returns nothing if there's no such tweak.
+ * Returns nil if there's no such tweak.
  */
 static int
 luaT_tweaks_index(struct lua_State *L)
 {
 	const char *name = luaL_checkstring(L, 2);
 	struct tweak *t = tweak_find(name);
-	if (t == NULL)
-		return 0;
+	if (t == NULL) {
+		lua_pushnil(L);
+		return 1;
+	}
 	struct tweak_value v;
 	tweak_get(t, &v);
 	return luaT_push_tweak_value(L, &v);
```

</details>